esmvaltool/diag_scripts/ipcc_ar5/ch09_fig09_17.py

```diff
@@ -0,0 +1,431 @@
+        else:
+            color = 'blue'
+
+
+        cube = cubedic[filename]
+
+        if 'MultiModel' in metadata[filename]['dataset']:
+            timeplot(
+                cube,
+                c=color,
+                # label=metadata[filename]['dataset'],
+                ls=':',
+                lw=2.,
+            )
+            plot_details[filename] = {
+                'c': color,
+                'ls': ':',
+                'lw': 2.,
+                'label': metadata[filename]['dataset']
+            }
+        else:
+            timeplot(
+                cube,
+                c=color,
+                # label=metadata[filename]['dataset'])
+                ls='-',
+                lw=2.,
+                title = 'Ocean heat content anomaly [J]/n(with respect to 1971)',
+                xlabel = 'Year',
+                ylabel = 'Joule',
+            )
+            plot_details[filename] = {
+                'c': color,
+                'ls': '-',
+                'lw': 2.,
+                'label': metadata[filename]['dataset'],
+                'title': 'Ocean heat content anomaly [J]/n(with respect to 1971)',
+                'xlabel': 'Year',
+                'ylabel': 'Joule',
+            }
+
+#            title = metadata[filename]['long_name']
+
+    # Saving files:
+    if cfg['write_plots']:
+        path = diagtools.get_image_path(
+            cfg,
+            metadata[filename],
+            prefix='MultipleModels_',
+            suffix='_'.join(['timeseries',
+                             str(layer) + image_extention]),
+            metadata_id_list=[
+                'field', 'short_name', 'preprocessor', 'diagnostic',
+                'start_year', 'end_year'
+            ],
+        )
+
+    # Resize and add legend outside the axes.
+    plt.gcf().set_size_inches(9., 6.)
+    diagtools.add_legend_outside_right(
+        plot_details, plt.gca(), column_width=0.15)
+
+    logger.info('Saving plots to %s', path)
+    plt.savefig(path)
+    plt.close()
+
+
+def global_sum(
+        cfg,
+        metadata,
+):
+    """
+    sum up the global heat content file to a 1-D time series
+
+    Parameters
+    ----------
+    cfg: dict
+        the opened global config dictioniary, passed by ESMValTool.
+    metadata: dict
+        The metadata dictioniary for a specific model.
+
+    """
+    cubedic={}
+    for filename in sorted(metadata):
+        cube = iris.load_cube(filename)
+        cube = diagtools.bgc_units(cube, metadata[filename]['short_name'])
+        cube = cube.collapsed(['longitude', 'latitude', 'depth'],iris.analysis.SUM)
+        cube = cube.aggregated_by('year', iris.analysis.MEAN)
+        cube.remove_coord('day_of_month')
+        cube.remove_coord('day_of_year')
+        cube.remove_coord('month_number')
+        #######
+        # subtract reference year value 1971
+        refvalue = cube.extract(iris.Constraint(year = 1971))
+        cube.data -= refvalue.data
```

Comment:
whereas `cube1.data - cube2.data` fully realizes the datas from both cubes, `cube1 - cube2` doesn't and is much lighter on memory (the latter one uses dask and lazy data). The end product is the same; you can subtract cubes if they have the same attributes and time axes which should be the case since refvalue is a subcube of cube

Reply:
@valeriupredoi yes, 

> should be the case

but isn't unfortunately. I've tried this with a similar instance in the code
https://github.com/ESMValGroup/ESMValTool/pull/922/files/b8477b62d9a7cbe321d61ec4eb1638da3cd09518#diff-d40c5497e38dc108f60f79b6810a86f9
`        # 2. multiply with each other and with cprho0
        # try working with cubes here to save memory
        cube *= volume * RHO_CP
#        cube.data *= volume.data * RHO_CP.data
#        cube.units *= volume.units * RHO_CP.units
        return cube  `

and here the multiplication of the cubes did not work because "This operation cannot be performed as there are differing coordinates (longitude, depth, cell index along first dimension, latitude, cell index along second dimension) remaining which cannot be ignored." 
This means broadcasting doesn't work here fro some reason.

Reply:
@ledm any idea on how to multiply a 4D cube with a 3D cube with a scalar cube? 
> cube *= volume * RHO_CP
I'm actually suprised that this doesn't work, perhaps the specific Iris version we use has an issue here? -> https://github.com/SciTools/iris/issues/2599

Reply:
I think we do this in `_area.py`, when the cube `cube` has 4 dimensions but the `grid_areas` has only three:
```
            if cube.data.ndim == 4 and grid_areas.ndim == 3:
                grid_areas = np.tile(grid_areas,
                                     [cube_shape[0], 1, 1, 1])
```
Hope this helps

Reply:
In that that isn't clear, we tile copies of the 3D data unit it has the same shape as the cube then we multiply them. It's possibly not the wisest way to do things, but I'd like to think iris can be lazy about it - but I haven't thought about it in great detail decently.

Reply:
@ledm thank you! but this means that indeed there is no numpy-style broadcasting with Iris cubes, even though it's a physically entire;y meaningful operation? I really have to manually build dummy cubes with the matching dimension? Sounds like a rather severe limitation (in Iris) to me ... :(

Reply:
the cube is collapsed on all its spatial coordinates anyway (@ledm have a look above) so it should work fine at the subtraction operation without tiling; @tillku you may want to remove the `year` attribute before subtracting, I think that's the bit that is preventing the operation; also looking a bit closed - you are subtracting a cube with one year (1971) out of a cube with N x years but not in a loop on years; this will work at array level but it has no meaning at cube level ie you want to subtract 1971 out of __each__ year from your N x years cube

Reply:
@valeriupredoi sorry for the confusion, in the example with 1971 that you linked above it actually works. I'll link to other instance where it doesn't work in a sec